pyproject.toml

```diff
@@ -13,7 +13,28 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">= 3.7"
+python = "^3.9"
+setuptools-scm = "3.3.3"
+numpy = "1.24.3"
+configparser = "^5.0.2"
+future = "0.17.1"
+futures = "3.1.1"
```

Comment:
@silverjam, now that we have dropped support for Python 2, do we still need these backports?

Not critical to do in this change. I can open a ticket if we want to punt for now.